### ISSUE TITLE
Bad Segmentation Result

### ISSUE THREAD

**Issue by GE-KI62** [Pointcept/PointTransformerV3#131]
Hello,

I have tried applying the pretrained PTV3 nuscenes model (https://huggingface.co/Pointcept/PointTransformerV3/tree/main/nuscenes-semseg-pt-v3m1-0-base) on my own data (only coord and intensity because nuscenes model takes 4 features as input). The result is disappointing. Am I doing something wrong or does the model need to be retrained to fit my distribution?

This is my custom preprocessing code to transform .las files:

```
def read_las(filepath):
    las = laspy.read(filepath)
    coords = las.xyz
    strength = las.intensity / 65535
    return coords, strength

def handle_process(
    scene_path, output_path):
    scene_id = os.path.basename(scene_path)

    coords, strength = read_las(scene_path)

    save_dict = dict(
        coord=coords.astype(np.float32),
        strength=strength.astype(np.float32).reshape(-1, 1)
    )

    # Save processed data
    os.makedirs(os.path.join(output_path, scene_id), exist_ok=True)

    for key in save_dict.keys():
        np.save(os.path.join(output_path, scene_id, f"{key}.npy"), save_dict[key])
```

Custom Dataset Wrapper:
```
@DATASETS.register_module()
class CustomDataset(DefaultDataset): 
        def __init__(self, **kwargs):
            super().__init__(ignore_index=-9999, **kwargs)
```

and my inference code:
```
data_config = dict(
        type='CustomDataset',
        split='test',
        data_root='data/laz_processed',
        transform=[
            dict(type="CenterShift", apply_z=True),
            dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength"),
                return_grid_coord=True,
            ),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord"),
                feat_keys=("coord", "strength"),
            )
        ],
        test_mode=False)

model_config = dict(
    type='DefaultSegmentorV2',
    num_classes=16,
    backbone_out_channels=64,
    backbone=dict(
        type='PT-v3m1',
        in_channels=4,
        order=['z', 'z-trans', 'hilbert', 'hilbert-trans'],
        stride=(2, 2, 2, 2),
        enc_depths=(2, 2, 2, 6, 2),
        enc_channels=(32, 64, 128, 256, 512),
        enc_num_head=(2, 4, 8, 16, 32),
        enc_patch_size=(1024, 1024, 1024, 1024, 1024),
        dec_depths=(2, 2, 2, 2),
        dec_channels=(64, 64, 128, 256),
        dec_num_head=(4, 4, 8, 16),
        dec_patch_size=(1024, 1024, 1024, 1024),
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        attn_drop=0.0,
        proj_drop=0.0,
        drop_path=0.3,
        shuffle_orders=True,
        pre_norm=True,
        enable_rpe=False,
        enable_flash=True,
        upcast_attention=False,
        upcast_softmax=False,
        cls_mode=False,
        pdnorm_bn=False,
        pdnorm_ln=False,
        pdnorm_decouple=True,
        pdnorm_adaptive=False,
        pdnorm_affine=True,
    pdnorm_conditions=("ScanNet", "S3DIS", "Structured3D"),
))

checkpoint = "./ptv3_checkpoints/nuscenes-semseg-pt-v3m1-0-base/model/model_best.pth"
keywords = "backbone."
import gc

if __name__ == "__main__":
    model = build_model(model_config).cuda()
    checkpoint = torch.load(
        checkpoint, map_location=lambda storage, loc: storage.cuda()
    )

    # Get model and checkpoint keys
    #model_keys = list(set(model.state_dict().keys()))

    weight = OrderedDict()
    for key, value in checkpoint["state_dict"].items():
        weight[key.replace("module.", "")] = value

    load_state_info = model.load_state_dict(weight, strict=True)

    dataset = build_dataset(data_config)

    with torch.no_grad():
        for i in range(len(dataset)):
            data = dataset[i]

            for key in data.keys():
                if isinstance(data[key], torch.Tensor):
                    data[key] = data[key].cuda(non_blocking=True)

            output = model(data)["seg_logits"]
            probs = torch.softmax(output, dim=1)
            labels = torch.argmax(probs, dim=1)

            import laspy
            import numpy as np

            header = laspy.LasHeader(point_format=0, version="1.2")
            header.scales = np.array([0.001, 0.001, 0.001])
            header.offsets = np.array([0, 0, 0])
            las = laspy.LasData(header)

            las.xyz = data["coord"].cpu().numpy()
            las.classification = labels.cpu().numpy()

            las.write(f"./output_{i}.las")

            del data, output, probs, labels
            gc.collect()
            torch.cuda.empty_cache()
            print(i, " finished")
```

Here is a comparison of the input vs output:

![Image](https://github.com/user-attachments/assets/0b4d0b2b-165e-490f-852f-6ee9a5e70765)

![Image](https://github.com/user-attachments/assets/0b6f1f7e-9f08-4003-9b44-c8e299227ce1)

Thanks in advance for any help!

**Comment by Gofinge** [comment]
I think the model weight trained by NuScene supports for Lidar pattern of point clouds, instead of such a dense and textured outdoor sceneria. For your case I suggest:
1. Label some data and train by yourself;
2. We will soon release a super-powerful SSL pretrained model weight. This weight should be work on your special data pattern.

**Comment by ccblublu** [comment]
@Gofinge i use the weights form huggingface to infer nuscenes data still get bad result
  
<img width="2094" height="1349" alt="Image" src="https://github.com/user-attachments/assets/0c840fd7-275a-45d3-92a5-a7b10af02bc7" />
